File: dir_rangan_playpen/test_fftw.c
// gcc -o test_fftw.out test_fftw.c -I/usr/local/include/ -L/usr/local/lib -lfftw3 -lm ; ./test_fftw.out ;

#include <fftw3.h>
#include <math.h>
#include <complex.h>
#include <stdlib.h>
#include <stdio.h>

int main()
{
  int verbose=2;
  int N=8,n=0;
  double pi=0,theta=0;
  fftw_complex *in, *out;
  fftw_plan p;
  if (verbose){ printf(" %% [entering test_fftw], N %d\n",N);}
  pi = 4.0*atan(1.0);
  in = (fftw_complex*) fftw_malloc(sizeof(fftw_complex) * N);
  for (n=0;n<N;n++){ theta = 2*pi*n/((double) N); in[n][0] = cos(theta); in[n][1] = sin(theta) ;}
  if (verbose>1){ printf(" %%%%  in: "); for (n=0;n<N;n++){ printf(" %+0.3f+i%+0.3f ",in[n][0],in[n][1]);} printf("\n");}
  out = (fftw_complex*) fftw_malloc(sizeof(fftw_complex) * N);
  p = fftw_plan_dft_1d(N, in, out, FFTW_FORWARD, FFTW_ESTIMATE);
  fftw_execute(p); /* repeat as needed */
  if (verbose>1){ printf(" %%%% out: "); for (n=0;n<N;n++){ printf(" %+0.3f+i%+0.3f ",out[n][0],out[n][1]);} printf("\n");}
  fftw_destroy_plan(p);
  fftw_free(in); fftw_free(out);
  if (verbose){ printf(" %% [finished test_fftw], N %d\n",N);}
  return 0;
}

File: dir_rangan_playpen/test_fftw.f
      program test_fftw
c$$$      gfortran -w -o test_fftw.out test_fftw.f -I/usr/include/ -L/usr/lib -lfftw3 ; ./test_fftw.out ;
      implicit none
      include '/usr/include/fftw3.f'
      integer n_fftw
      parameter (n_fftw=8)
      integer na
      double complex fftw_in1_(0:n_fftw-1)
      double complex fftw_in2_(0:n_fftw-1)
      double complex fftw_out_(0:n_fftw-1)
      real *8 pi,theta
      integer*8 plan_backward
      integer*8 plan_forward      
      pi = 4.0*atan(1.0)
      do na = 0,n_fftw-1
         theta = 1*na*2*pi/n_fftw
         fftw_in1_(na) = dcmplx(cos(theta),sin(theta))
      end do
      write (6,'(A)') 'fftw_in1_:'
      write (6,'(2F8.3)') (fftw_in1_(na),na=0,n_fftw-1)
      call dfftw_plan_dft_1d_(plan_forward,n_fftw,fftw_in1_,fftw_out_,
     $     FFTW_FORWARD,FFTW_ESTIMATE)
      call dfftw_execute_(plan_forward)
      write (6,'(A)') 'fftw_out_:'
      write (6,'(2F8.3)') (fftw_out_(na),na=0,n_fftw-1)
      call dfftw_plan_dft_1d_(plan_backward,n_fftw,fftw_out_,fftw_in2_,
     $     FFTW_BACKWARD,FFTW_ESTIMATE)
      call dfftw_execute_(plan_backward)
      write (6,'(A,I0)') 'fftw_in2_ / ',n_fftw
      write (6,'(2F8.3)') (fftw_in2_(na)/n_fftw,na=0,n_fftw-1)
      call dfftw_destroy_plan_(plan_forward)
      call dfftw_destroy_plan_(plan_backward)
      stop
      end

      include 'dfftpack.f'
